### DISCUSSION TITLE
What's the use case/feature difference between `vtm` and `tmux`?

### DISCUSSION THREAD

**Discussion by jdrch** [directvt/vtm#499] [General]
Dumb question, but I'm curious about the above. I've always struggled understand how `tmux` works but understood `vtm` pretty quickly. However, `tmux` is readily available in various repos while `vtm` isn't. Ergo, while `vtm` may be easier to use, keeping it updated is much more difficult than and doesn't scale as well as keeping `tmux` updated. So basically I'm wondering if `vtm` is worth that difficulty relative to `tmux` based on its features relative to `tmux`.

**Comment by o-sdn-o** [comment]
Thank you for the very good questions.

vtm is just beginning to be applicable in workflows (maybe this is not entirely true, it just seems so to me). It is not stable in the UI, in the configuration parameters, in the keyboard bindings, in the binary structure of the protocols used, etc. In addition, very often regressions occur in subsequent releases that make it impossible to use. This is due to the fact that I do not have the opportunity to deeply test each release, as this will negatively affect the pace of development, which, as it seems to me, is not moving fast enough. When the main functionality of vtm takes on clear outlines, testing of each release will be more thorough.

As for the differences with tmux.
tmux relies primarily on the keyboard.
vtm is currently mouse-based, making vtm's functionality fairly intuitive and easy to navigate with a mouse, but full keyboard support is required for productive day-to-day work. Keyboard support in vtm is in development, this is Milestone v1.0.0. I will make it possible to operate inside vtm relying solely on the keyboard.

Another big difference from tmux is that the vtm environment provides the launch of a special type of application in DirectVT mode, which are neither graphical (although very close to this type) nor console-based (although they can run in any text console outside vtm). Examples of such applications are `vtm -r term`, `vtm -r gems`, `vtm -r calc`, `vtm -r truecolor` and `vtm -r text`. They can be compiled as standalone executables. Once vtm becomes stable, a separate framework for creating applications of this type will be published. By the way, the vtm desktop itself is an application of this type.

Regarding updates, to ensure a continuous update process, vtm must be registered for inclusion in the required distribution's package, having first met a number of requirements (for example, for inclusion in Debian https://www.debian.org/doc/manuals/maint-guide/) . This is tracked in #77. This will be done after the development of the main functionality is completed. For now, you can only update after implementing a specific feature or fixing a specific bug that you are waiting for to be fixed. I try to indicate what specific bug or feature I'm working on at the moment. If a bug I know crashes vtm or something else, then I work on it first. In general, I try to fix bugs first and then implement functionality.

**Reply by jdrch** [reply]
Makes sense, thanks! I was pretty excited when it worked just fine on my work PC without admin privileges. Looking forward to seeing it in repos!
